File: frontend/types/learning-path.ts
export interface LearningPathItem {
  id: string
  path_id: string
  knowledge_id?: string | null
  wiki_page_id?: string | null
  title: string
  item_type: string
  order_index: number
  status: "pending" | "doing" | "completed" | "skipped"
  reason?: string | null
  estimated_minutes?: number | null
  completed_at?: string | null
  created_at: string
}

export interface LearningPath {
  id: string
  user_id: string
  course_id: string
  title: string
  goal?: string | null
  reason?: string | null
  status: string
  progress: number
  strategy_version_id?: string | null
  created_at: string
  updated_at: string
  items: LearningPathItem[]
}

export interface LearningPathGeneratePayload {
  course_id: string
  goal?: string
  target_knowledge_ids?: string[]
  path_type?: string
}

File: backend/app/schemas/learning_path.py
from __future__ import annotations

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field


class LearningPathGenerateRequest(BaseModel):
    course_id: UUID
    goal: str = Field(default="补强当前薄弱知识点", min_length=1, max_length=500)
    target_knowledge_ids: list[UUID] = Field(default_factory=list)
    path_type: str = Field(default="weakness_repair", max_length=64)


class LearningPathItemUpdate(BaseModel):
    status: str = Field(pattern="^(pending|doing|completed|skipped)$")


class LearningPathItemRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    path_id: UUID
    knowledge_id: UUID | None = None
    wiki_page_id: UUID | None = None
    title: str
    item_type: str
    order_index: int
    status: str
    reason: str | None = None
    estimated_minutes: int | None = None
    completed_at: datetime | None = None
    created_at: datetime


class LearningPathRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    user_id: UUID
    course_id: UUID
    title: str
    goal: str | None = None
    reason: str | None = None
    status: str
    progress: float = 0
    strategy_version_id: UUID | None = None
    created_at: datetime
    updated_at: datetime
    items: list[LearningPathItemRead] = Field(default_factory=list)
